be/src/exec/pipeline/scan/morsel.h

```diff
@@ -223,19 +235,76 @@ class FixedMorselQueue final : public MorselQueue {
     std::atomic<size_t> _pop_index;
 };
 
-class PhysicalSplitMorselQueue final : public MorselQueue {
+class SplitMorselQueue : public MorselQueue {
 public:
-    explicit PhysicalSplitMorselQueue(Morsels&& morsels, int64_t degree_of_parallelism, int64_t splitted_scan_rows)
+    SplitMorselQueue(Morsels&& morsels, int64_t degree_of_parallelism, int64_t splitted_scan_rows)
             : _morsels(std::move(morsels)),
               _num_original_morsels(_morsels.size()),
               _degree_of_parallelism(degree_of_parallelism),
               _splitted_scan_rows(splitted_scan_rows) {}
+    void set_tablets(const std::vector<TabletSharedPtr>& tablets) override { _tablets = tablets; }
+
+    void set_ticket_checker(const query_cache::TicketCheckerPtr& ticket_checker) { _ticket_checker = ticket_checker; }
+
+protected:
+    template <typename T>
+    auto _use_ticket_checker(std::function<T(query_cache::TicketCheckerPtr& ticket_checker, int64_t)> func,
+                             T default_value) {
+        if (_ticket_checker == nullptr) {
+            return default_value;
+        } else {
+            DCHECK(0 <= _tablet_idx && _tablet_idx < _tablets.size());
+            auto tablet_id = _tablets[_tablet_idx]->tablet_id();
+            return func(_ticket_checker, tablet_id);
+        }
+    }
+
+    int _inc_num_splits_unlocked() {
```

Comment:
why _inc_num_splits_unlocked has the suffix _unlocked but _emit_eos_morsel and _has_emitted_eos_morsel without?

Reply:
unlocked suffix is meanless, so it is removed.